File: JAVA DSA/SORTING_ALGOS/BubbleSort.java
package SORTING_ALGOS;

public class BubbleSort {
    public static void main(String[] args) {
        int [] arr = {1,2,4,1,9,7};
        
        for(int i=arr.length-1; i>=1;i--){
            int didswap=0;
            for(int j=0;j<=i-1;j++){
                if(arr[j] > arr[j+1]){
                    int temp=arr[j];
                    arr[j]=arr[j+1];
                    arr[j+1]= temp;
                    didswap = 1;
                }
            }
            if(didswap == 0) break;     //optimised for best case, in case all sorted
        }
        for(int ele:arr)System.out.print(ele + " ");

    }
}

File: C PROGRAMS/DAA 3rd Sem/BubbleSort.c
#include<stdio.h>

void bubbleSort(int arr[],int n){

    for(int i=0;i<n-1;i++){
        for(int j=0;j<n-i-1;j++){
            if(arr[j] > arr[j+1]){
                int temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
    }

}

int main(){

    int arr[]={10,9,8,7,6,5,4,3,2,1};
    int n = 10;

    bubbleSort(arr,n);

    for(int i=0;i<n;i++){
        printf("%d ",arr[i]);
    }

    return 0;
}